docs/source/notes/resumption.rst

```diff
@@ -0,0 +1,134 @@
+
+1. Local checkpoints of the format ``"{save_folder}/latest-rank0"``. The format for the latest checkpoint can be configured with ``save_latest_filename`` argument (default: ``latest-rank{rank}``).
+2. If no local checkpoints are found, then each logger is checked for files of the format ``"{run_name}/checkpoints/latest-rank{rank}"``. This is often used for resuming from an object store such as S3.
+3. Finally, ``load_path`` is used to load a checkpoint. This can be used for example, a fine-tuning run on a spot instance, where ``load_path`` would be set to the original weights.
+
+Example: Object Store
+---------------------
+
+A typical use case is saving checkpoints to object store (e.g. S3) when there is no local file storage shared across runs. For example, a setup such as this:
+
+
+.. testcode::
+
+    from composer.loggers import ObjectStoreLogger
+
+    object_store_logger = ObjectStoreLogger(
+        object_store_cls=LibcloudObjectStore,
+        object_store_kwargs={
+            "provider": "s3",
+            "container": "checkpoint-debugging",  # bucket name
+            "provider_kwargs": {
+                'key': 'provider_key',  # The cloud provider key.
+                'secret': '*******',  # The cloud provider secret.
+            },
+        },
+    )
+
+    trainer = Trainer(
+        ...,
+        autoresume=True,
+        save_folder='checkpoints',
+        save_num_checkpoints_to_keep=0,  # delete all checkpoints locally
+        run_name='my_cool_run',
+        save_artifact_name='checkpoints/ep{epoch}.pt',
+        loggers=[object_store_logger],
+    )
+
+    trainer.fit()
+
+
+During resumption, there would be no local checkpoints, so the trainer would then look in the object store logger's provided bucket and artifact folder (`checkpoint-debugging/my_cool_run/checkpoints`) to find the latest checkpoint.
+
+Example: Fine-tuning
+--------------------
+
+To run fine-tuning on a spot instance, ``load_path`` would be set to the original weights and an object store logger would be added.
+
+.. testsetup::
+
+    from composer.loggers import ObjectStoreLogger
+
+    object_store_logger = ObjectStoreLogger(
+        object_store_cls=LibcloudObjectStore,
+        object_store_kwargs={
+            "provider": "s3",
+            "container": "checkpoint-debugging",  # bucket name
+            "provider_kwargs": {
+                'key': 'provider_key',  # The cloud provider key.
+                'secret': '*******',  # The cloud provider secret.
+            },
+        },
+    )
+
+.. testcode::
+
+    trainer = Trainer(
+        ...,
+        autoresume=True,
+        load_path='pretrained_weights/model.pt',
+        save_path='checkpoints',
```

Comment:
Would it be helpful to include the "load_object_store" argument to show that the pretrained weights could be stored in a bucket too?

Reply:
I added a note about this, so as not to clutter up this signature too much.